src/effects/aura.c

```diff
@@ -1,11 +1,194 @@
 #include "common.h"
 #include "effects_internal.h"
 
+extern Gfx D_090023E0[];
+extern Gfx D_09002440[];
+extern Gfx D_090023F8[];
+extern Gfx D_09002458[];
+extern Gfx D_090023B0[];
+extern Gfx D_09002410[];
+extern Gfx D_090023C8[];
+extern Gfx D_09002428[];
+extern Gfx D_09002000[];
+extern Gfx D_090020E8[];
+extern Gfx D_090021D0[];
+
+Gfx* D_E0076E90[] = { D_090023E0, D_090023E0, D_09002440, D_09002440 };
+Gfx* D_E0076EA0[] = { D_090023F8, D_090023F8, D_09002458, D_09002458 };
+Gfx* D_E0076EB0[] = { D_090023B0, D_090023B0, D_09002410, D_09002410 };
+Gfx* D_E0076EC0[] = { D_090023C8, D_090023C8, D_09002428, D_09002428 };
+Gfx* D_E0076ED0[] = { D_09002000, D_09002000, D_090020E8, D_090021D0 };
+
 void aura_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/aura", aura_main);
+typedef struct AuraFXData {
+    /* 0x00 */ s32 unk_00;
+    /* 0x04 */ Vec3f pos;
+    /* 0x10 */ f32 unk_10;
+    /* 0x14 */ f32 unk_14;
+    /* 0x18 */ f32 unk_18;
+    /* 0x1C */ f32 unk_1C;
+    /* 0x20 */ f32 unk_20;
+    /* 0x24 */ f32 unk_24;
+    /* 0x28 */ s32 unk_28;
+    /* 0x2C */ s32 unk_2C;
+    /* 0x30 */ s32 unk_30;
+    /* 0x34 */ f32 unk_34;
+    /* 0x38 */ f32 unk_38;
+    /* 0x3C */ f32 unk_3C;
+    /* 0x40 */ f32 unk_40;
+    /* 0x44 */ f32 unk_44;
+    /* 0x48 */ f32 unk_48;
+    /* 0x4C */ f32 unk_4C;
+    /* 0x50 */ f32 unk_50;
+    /* 0x54 */ f32 unk_54;
+    /* 0x58 */ f32 unk_58;
+    /* 0x5C */ f32 unk_5C;
+    /* 0x60 */ f32 unk_60;
+               union {
+                   s32 s;
+                   f32 f;
+               } unk_64;
+    /* 0x68 */ u8 unk_68;
+    /* 0x69 */ u8 unk_69;
+    /* 0x6A */ u8 unk_6A;
+    /* 0x6B */ u8 unk_6B;
+    /* 0x6C */ u8 unk_6C;
+    /* 0x6D */ u8 unk_6D;
+    /* 0x6E */ u8 unk_6E;
+} AuraFXData; // size = 0x70
+
+void aura_init(EffectInstance* effect);
+void aura_update(EffectInstance* effect);
+void aura_render(EffectInstance* effect);
+
+void aura_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, EffectInstance** outEffect) {
+    EffectBlueprint bp;
+    EffectBlueprint* bpPtr = &bp;
+    EffectInstance* effect;
+    AuraFXData* part;
+    s32 numParts = 1;
+    f32 zero;
+
+    bp.init = aura_init;
+    bp.update = aura_update;
+    bp.renderWorld = aura_render;
+    bp.unk_00 = 0;
+    bp.unk_14 = 0;
+    bp.effectID = EFFECT_AURA;
+
+    effect = shim_create_effect_instance(bpPtr);
+    effect->numParts = numParts;
+    
+    part = effect->data = shim_general_heap_malloc(numParts * sizeof(*part));
+
+    ASSERT(effect->data != NULL);
+
+    part->unk_64.s = 0;
+    zero = part->unk_64.f;
+    part->unk_40 = zero;
```

Comment:
I'm thinking you can get away with using 0 as a literal by doing a chained assignment (thing = other = stuff = 0)..worth a shot

Reply:
It's probably not going to work
![image](https://user-images.githubusercontent.com/50085307/155700285-e26aab76-d13e-4f86-b76e-a18060ddf25f.png)